runners/flink/src/test/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperatorTest.java

```diff
@@ -432,7 +432,7 @@ public void testEnsureStateCleanupWithKeyedInputCleanupTimer() {
     cleanupTimer.setForWindow(KV.of("key", "string"), window);
 
     Mockito.verify(stateBackendLock).lock();
-    ByteBuffer key = FlinkKeyUtils.encodeKey("key", keyCoder, Coder.Context.NESTED);
+    ByteBuffer key = FlinkKeyUtils.encodeKey("key", keyCoder);
```

Comment:
```suggestion
    ByteBuffer key = FlinkKeyUtils.encodeKey("key", keyCoder, Coder.Context.NESTED);
```

Needs to be reverted.

Reply:
Thanks @mxm for the review. Reverting this will cause `Key Group xx does not belong to the local range.` exception. I think somewhere in the source there's a mismatch between encoding methods. Do we have a test for state cleanup?

Reply:
Yes, there is a test which you also modified :) I'm extending it here: #8472. We also realized there is something odd about the state request key encoding. It's because of the wrapping via `LengthPrefixCoder` for the keyCoder.

Reply:
Confirm that #8472 also fixes this key encoding issue. I'll revert this change. Thanks @mxm.

Reply:
Great!